src/include/homestore/btree/detail/btree_mutate_impl.ipp

```diff
@@ -203,7 +203,12 @@ btree_status_t Btree< K, V >::mutate_write_leaf_node(const BtreeNodePtr& my_node
                 BT_DBG_ASSERT(false, "For non-extent keys range-update should be really update and cannot insert");
                 ret = btree_status_t::not_supported;
             } else {
+                const auto new_val_size{(*req.m_newval).serialized_size()};
```

Comment:
What will happen if start_found or end_found is not true?

Reply:
When we are in this point, we are sure that we are in the leaf. Before calling this function the request is getting trimmed `req.trim_working_range( my_node->min_of(` for the leaf. 
For the out of band case, if we point to same start and end without any match, it is hitting unavailable range. In this case, `count = my_node->template get_all< K, V > `returns 0 and `BT_NODE_LOG_ASSERT` would be raised accordingly.

Reply:
Can we add below test case with rangeput, the expected result should be RangePut should fail. 
Case-1:  start_key is invalid, end_key is valid.
Case-2: start_key is valid, end_key is invalid.
Case-3: both keys are invalid.